core/sail/lmdb/src/main/java/org/eclipse/rdf4j/sail/lmdb/LmdbSailStore.java

```diff
@@ -573,6 +570,51 @@ public void approve(Resource subj, IRI pred, Value obj, Resource ctx) throws Sai
 			addStatement(subj, pred, obj, explicit, ctx);
 		}
 
+		@Override
+		public void approveAll(Set<Statement> approved, Set<Resource> approvedContexts) {
+
+			sinkStoreAccessLock.lock();
+			try {
+				startTransaction(true);
+
+				for (Statement statement : approved) {
+					Resource subj = statement.getSubject();
+					IRI pred = statement.getPredicate();
+					Value obj = statement.getObject();
+					Resource context = statement.getContext();
+
+					AddQuadOperation q = new AddQuadOperation();
+					q.s = valueStore.storeValue(subj);
+					q.p = valueStore.storeValue(pred);
+					q.o = valueStore.storeValue(obj);
+					q.c = context == null ? 0 : valueStore.storeValue(context);
+					q.context = context;
+					q.explicit = explicit;
+
+					if (multiThreadingActive) {
+						while (!opQueue.add(q)) {
+							if (tripleStoreException != null) {
+								throw wrapTripleStoreException();
+							}
+						}
+
```

Comment:
Thread.yield() is missing here...
Could be fixed by just re-using `addStatement`